core/src/main/java/org/kestra/core/tasks/scripts/Bash.java

```diff
@@ -99,38 +112,101 @@
         },
         dynamic = true
     )
-    private List<String> files;
+    protected List<String> outputs;
+
+    @InputProperty(
+        description = "Input files are extra files supplied by user that make it simpler organize code.",
+        body = {
+            "Describe a files map that will be written and usable in execution context. In python execution context is in a temp folder, for bash scripts, you can reach files using a scriptFolder variable like 'source {{scriptFolder}}/myfile.sh' "
+        },
+        dynamic = true
+    )
+    protected HashMap<String, String> inputFiles;
+
+    protected volatile List<File> cleanupDirectory;
+    protected volatile String tmpFolder;
+
 
     @Override
     public Bash.Output run(RunContext runContext) throws Exception {
+        tmpFolder = "/tmp/" + UUID.randomUUID();
+        return run(runContext, throwFunction((tempFiles) -> {
+            // final command
+            List<String> renderer = new ArrayList<>();
+
+            if (this.exitOnFailed) {
+                renderer.add("set -o errexit");
+            }
+
+            // renderer command
+            for (String command : this.commands) {
+                renderer.add(runContext.render(
+                    command,
+                    tempFiles.size() > 0 ? ImmutableMap.of("temp", tempFiles, "scriptFolder", tmpFilesFolder()) : ImmutableMap.of("scriptFolder", tmpFilesFolder())
+                ));
+            }
+
+            return String.join("\n", renderer);
+        }));
+    }
+
+    protected void handleInputFiles(RunContext runContext) throws IOException, IllegalVariableEvaluationException {
+        if (inputFiles != null && inputFiles.size() > 0) {
+            File tmpFileFolderHandler = new File(tmpFilesFolder());
+            if (tmpFileFolderHandler.exists()) {
+                FileUtils.deleteDirectory(tmpFileFolderHandler);
+            }
+            Files.createDirectories(Paths.get(tmpFilesFolder()));
+
+            cleanupDirectory.add(tmpFileFolderHandler);
+            cleanupDirectory.add(new File(tmpFolder));
+
+            for (String fileName : inputFiles.keySet()) {
+                String subFolder = tmpFilesFolder() + "/" + new File(fileName).getParent();
```

Comment:
Seems that you don't support multiple directory like `folder/Aa/bb`, we must :smile: